### ISSUE TITLE
Connecting through zero trust access proxies

### ISSUE THREAD

**Issue by aniqueta** [nextcloud/ios#1565]
### Steps to reproduce

1. Place Nextcloud server address behind zero trust access proxy like Cloudflare Access, Pomerium, or Pritunl
2. Attempt to setup account in iOS app; enter serve address normally, e.g., https://my.nextcloud.org

### Expected behaviour

The iOS app shows the web-based authentication page  for the access proxy, and after passing through that access proxy, it caches the authentication tokens, and the Nextcloud authentication pops up (or does not if it's configured for SSO). By caching the authentication tokens, normal usage of the app is allowed until they expire. When they expire, the app throws up the access proxy authentication page to refresh the access tokens. 

### Actual behaviour

After entering the server address and tapping log in, the iOS app produces this error "Connection error 200: Transfer stopped." 

### Reasoning or why should it be changed/implemented?

This makes a zero trust configuration impossible if wanting to allow usage of the iOS app by the server's users. This will be increasingly important for enterprise users.

### Environment data

**Nextcloud iOS app version:** 3.3.1.1

**Comment by 0dragosh** [comment]
The ability to set a client side header that can use a service token would be a great alternative. 

Or, cloudflare only, basic auth credential input client side and a cloudflare worker that converts basic auth to a service token.

We need _something_ for this.

**Comment by SeaniedIRE** [comment]
Yes I have deployed zero trust on all my apps since its free so why the hell not! I get the same 200 error with nextcloud sadly. I was looking for an alternative, maybe auth with a cert to bypass the cloudflare proxy but no luck. This would increase security massively

**Comment by 0dragosh** [comment]
> Yes I have deployed zero trust on all my apps since its free so why the hell not! I get the same 200 error with nextcloud sadly. I was looking for an alternative, maybe auth with a cert to bypass the cloudflare proxy but no luck. This would increase security massively

mTLS for Cloudflare access is not free, FYI.

**Comment by fabriziosalmi** [comment]
While my Nextcloud web UI is working under Cloudflare zero-trust the mobile/desktop apps received a 403 when trying to adding the Nextcloud server.

**Comment by attzonko** [comment]
This is the same error seen when a nextcloud instance is sitting behind Traefik reverse proxy with an authentication forward service like Google Oauth2.  

Accessing the nextcloud from a web browser works fine where the user gets redirected to authenticate with the Oauth2 provider and gets redirected back to the instance.

Is this something planned to get fixed or functionality added to support these use cases of self hosting?

**Comment by SolarCzar** [comment]
I'm having the same problem.  I've tried modifying 'overwriteprotocol' and 'overwritehost' in config.php, as some of others on the web have been indicating, but with no joy.  Access via a browser (even on iOS) works fine, but the Nextcloud companion app errors out with NSURLError -999...

![IMG_9409 2](https://user-images.githubusercontent.com/66656008/178125901-2c83ac35-232f-4fc1-b485-a2168835cbe2.jpg)

**Comment by phantomski77** [comment]
I second this. A lot of us are running NexCloud on non-dedicated local infrastructure and we need some sort of Access Policy to protect it, instead of just opening the firewall port forwarding and relying on internal security.
Service authentification token would be a good start and a low hanging fruit. Easy to implement, easy to setup.

**Comment by laszlo462** [comment]
Another vote for addressing this issue.  Have migrated my self-hosted apps to Cloudflare tunnels using Google Oauth, app is now useless since it doesn't account for an auth redirect prior to landing at the login page.

**Comment by aiohdfgaiuhg** [comment]
bumping again

**Comment by B0lg0r0v** [comment]
bumping again

**Comment by KaiHicks** [comment]
I am running into the same issue, but I have NextCloud behind a reverse proxy, and I am protecting it using forward auth with my SSO provider. This way, unauthenticated attackers can't even ping NextCloud (or any other service for that matter), reducing the attack surface. Being able to set a JWT would be a great feature.

**Comment by Section82** [comment]
this would be great to have so I dont have to expose my nextcloud instance directly and could tunnel it through cloudflare like my other apps. Even the ability to put in an HTTP header so we could make a service token would be great!

**Comment by thimplicity** [comment]
Anything new on this one? facing the same issue

**Comment by 0xLun** [comment]
Same problem, it would be a great idea to be able to add an Header to authenticate on the proxy or the zero trust solution that protect the nextcloud instance.
